src/lib/dnssd/Discovery_ImplPlatform.cpp

```diff
@@ -54,34 +54,39 @@ static void HandleNodeResolve(void * context, DnssdService * result, CHIP_ERROR
 
     DiscoveredNodeData nodeData;
 
-    Platform::CopyString(nodeData.hostName, result->mHostName);
-    Platform::CopyString(nodeData.instanceName, result->mName);
+    Platform::CopyString(nodeData.hostName, results.data()[0].mHostName);
+    Platform::CopyString(nodeData.instanceName, results.data()[0].mName);
+    nodeData.port = results.data()[0].mPort;
 
-    if (result->mAddress.HasValue())
+    size_t addressesFound = 0;
+    for (auto & result : results)
     {
-        nodeData.ipAddress[0] = result->mAddress.Value();
-        nodeData.interfaceId  = result->mInterface;
-        nodeData.numIPs       = 1;
-    }
-    else
-    {
-        nodeData.numIPs = 0;
-    }
-
-    nodeData.port = result->mPort;
+        if (result.mAddress.HasValue())
+        {
+            if (addressesFound == ArraySize(nodeData.ipAddress))
+            {
+                // Out of space.
+                break;
+            }
+            nodeData.ipAddress[addressesFound] = result.mAddress.Value();
+            nodeData.interfaceId               = result.mInterface;
```

Comment:
does this change the interface in result?

What if result contains "IP: 1.2.3.4, Interface 1" and "IP: 5.6.7.8 Interface 2"?
This seems to create NodeData with "Interface: 2" and both IPs, which would be wrong.

Reply:
The API documentation says that the platform must not pass in items with different interface ids.

The Darwin implementation certainly does not.

We could check for that here, if you prefer: set the interface from the first result and drop the results that have a different interface.

Reply:
Done by changing the API so this is a non-issue.